internal/manifests/collector/ports.go

```diff
@@ -260,3 +279,10 @@ func portFromEndpoint(endpoint string) (int32, error) {
 
 	return int32(port), err
 }
+
+func isDuplicatePort(portsMap map[string]corev1.ContainerPort, servicePort corev1.ServicePort) bool {
+	for _, containerPort := range portsMap {
+		return containerPort.Protocol == servicePort.Protocol && containerPort.ContainerPort == servicePort.Port
```

Comment:
Isn't this going to just return for the first port in the map? This doesn't seem like it's going to search the map.

Reply:
good catch ! updated